modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -1633,7 +1633,7 @@ bool CvCapture_FFMPEG::retrieveFrame(int flag, unsigned char** data, int* step,
             img_convert_ctx,
             sw_picture->data,
             sw_picture->linesize,
-            0, context->coded_height,
```

Comment:
Lets use `sw_picture->height` here.

According to ffmpeg sample code: https://github.com/FFmpeg/FFmpeg/blob/n4.4.2/fftools/ffplay.c#L925-L926

---

BTW, from the same samples code there is no usage of `coded_width`/`coded_height` values with modern FFmpeg.
Perhaps we need to update `sws_getCachedContext()` call too (probably later, and align with "n0,8" ffmpeg version).

Reply:
Actually I tried 3.4 as it is and it works like a charm. No segfault.
Even the issue https://github.com/opencv/opencv/pull/22145 is not present.
Maybe there are some specific 4.x changes that brakes it.

Reply:
Code is different, but there is the same problem on 3.4: https://github.com/opencv/opencv/blob/3.4.18/modules/videoio/src/cap_ffmpeg_impl.hpp#L1409-L1411

Reply:
My bad. I don't know why but from time to time it pass without crashing.
But 3.4 can crash too.
I'll update the PR.

Reply:
PR ready for 3.4 (builds and run correctly)